zerver/lib/actions.py

```diff
@@ -5900,3 +5900,61 @@ def do_delete_realm_export(user_profile: UserProfile, export: RealmAuditLog) ->
     export.extra_data = ujson.dumps(export_data)
     export.save(update_fields=['extra_data'])
     notify_realm_export(user_profile)
+
+def get_topic_messages(
+    user_profile: UserProfile,
+    stream: Stream,
+    topic_name: str
+) -> List[Message]:
+    log_statsd_event('get_topic_messages')
+    messages = UserMessage.objects.filter(
+        user_profile=user_profile
+    )
+
+    recipient = stream.recipient
+    messages = messages.filter(message__recipient=recipient)
+
+    messages = filter_by_topic_name_via_message(
+        query=messages,
+        topic_name=topic_name,
+    )
+
+    return messages
+
+def do_move_topic_messages(
+    user_profile: UserProfile,
+    stream_origin: Stream,
+    stream_destiny: Stream,
+    topic_name: str
+) -> None:
+    # if they are not public streams don't move any messages
+    if not (stream_origin.is_public() and stream_destiny.is_public()):
+        return
+
+    user_messages = get_topic_messages(user_profile, stream_origin, topic_name)
+    recipient_destiny = get_stream_recipient(stream_destiny.id)
+
+    # We could use a bot for the breadcrump message too
```

Comment:
Yeah, let's change this to a message sent by Notification Bot.

Reply:
@mateuszmandera I just noticed we have these `get_stream_recipient` and `get_personal_recipient` helper functions in a few places that this code is copying; we should just remove those and use `stream.recipient` etc., right?  (And delete `get_recipient` and its cache as well, I think???)

In any case, for this PR, @dobleuber you can just use `new_stream.recipient`.

Reply:
@timabbott Yeah, we still have a bunch of places where we use those functions that haven't been changed. It should be pretty quick to do though and then we can remove the functions completely. I'll add it to my list.